lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1598 @@
+
+
+def handle_elastigroup(client, module):
+    has_changed = False
+    should_create = False
+    group_id = None
+    message = 'None'
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+    uniqueness_by = module.params.get('uniqueness_by')
+    external_group_id = module.params.get('id')
+
+    if uniqueness_by == 'id':
+        if external_group_id is None:
+            should_create = True
+        else:
+            should_create = False
+            group_id = external_group_id
+    else:
+        groups = client.get_elastigroups()
+        group_found, group_id = find_group_with_same_name(groups, name)
+        if group_found is True:
+            should_create = False
+        else:
+            should_create = True
+
+    if should_create is True:
+        if state == 'present':
+            eg = expand_elastigroup(module, is_update=False)
+
+            group = client.create_elastigroup(group=eg)
+            group_id = group['id']
+            message = 'Created group Successfully.'
+            has_changed = True
+
+        elif state == 'absent':
+            message = 'Cannot delete non-existent group.'
+            has_changed = False
+    else:
+        eg = expand_elastigroup(module, is_update=True)
+
+        if state == 'present':
+            group = client.update_elastigroup(group_update=eg, group_id=group_id)
+            message = 'Updated group successfully.'
+
+            try:
+                roll_config = module.params.get('roll_config')
+                if roll_config:
+                    eg_roll = spotinst.aws_elastigroup.Roll(
+                        batch_size_percentage=roll_config.get('batch_size_percentage'),
+                        grace_period=roll_config.get('grace_period'),
+                        health_check_type=roll_config.get('health_check_type')
+                    )
+                    roll_response = client.roll_group(group_roll=eg_roll, group_id=group_id)
+                    message = 'Updated and started rolling the group successfully.'
+            except SpotinstClientException as exc:
+                message = 'Updated group successfully, but failed to perform roll. Error:' + str(exc)
+            has_changed = True
+
+        elif state == 'absent':
+            try:
+                client.delete_elastigroup(group_id=group_id)
+            except SpotinstClientException as exc:
+                if "GROUP_DOESNT_EXIST" in exc.message:
+                    pass
+                else:
+                    module.fail_json(msg="Error while attempting to delete group : " + exc.message)
+
+            message = 'Deleted group successfully.'
+            has_changed = True
+
+    return group_id, message, has_changed
+
+
+def retrieve_group_instances(client, module, group_id):
+    wait_timeout = module.params.get('wait_timeout')
+    wait_for_instances = module.params.get('wait_for_instances')
+
+    if wait_timeout is None:
+        wait_timeout = 300
```

Comment:
In the integration tests related to RDS I have had to increase timeouts for creation to up to 800.  I had some similar problems working on the auto scaling groups earlier.  I would propose that this timeout should be at the level where the wait is sufficent 95% of the time even at times when AWS is reasonably busy and instance start up takes lots of time.  Do you know if 300 is long enough?

Reply:
Yeah, for spot instances I'd say 300 is a bit short, unless spotinst behaves *significantly* differently from the AWS spot instance market.